src/main/java/uk/gov/hmcts/reform/sscs/service/evidence/EvidenceUploadService.java

```diff
@@ -139,6 +143,29 @@ public boolean submitHearingEvidence(String identifier, EvidenceDescription desc
                 .orElse(false);
     }
 
+    public boolean deleteDraftHearingEvidence(String identifier, String evidenceId) {
+        return deleteEvidence(identifier, evidenceId, draftHearingDocumentExtractor);
+    }
+
+    private <E> boolean deleteEvidence(String identifier, String evidenceId, DocumentExtract<E> documentExtract) {
+        return onlineHearingService.getCcdCaseByIdentifier(identifier)
+                .map(caseDetails -> {
+                    List<E> documents = documentExtract.getDocuments().apply(caseDetails.getData());
+
+                    if (documents != null) {
+                        List<E> newDocuments = documents.stream()
+                                .filter(corDocument -> !documentExtract.findDocument().apply(corDocument).getDocumentLink().getDocumentUrl().endsWith(evidenceId))
+                                .collect(toList());
+                        documentExtract.setDocuments().accept(caseDetails.getData(), newDocuments);
+
+                        ccdService.updateCase(caseDetails.getData(), caseDetails.getId(), UPLOAD_DRAFT_DOCUMENT.getCcdType(), "SSCS - evidence deleted", UPDATED_SSCS, idamService.getIdamTokens());
+
+                        documentManagementService.delete(evidenceId);
```

Comment:
Out of interest: does this hard delete the document?

Reply:
And if documentManagementService.delete(evidenceId) fails, does this method return true still?

Reply:
Yes it will delete only draft document which user has uploaded
If it failed then on frontend user will see generic error message